client/utils/use-confirm-navigation.js

```diff
@@ -13,18 +13,13 @@ import { getHistory } from '@woocommerce/navigation';
  * - const callback = useConfirmNavigation( () => 'Are you sure you want to leave?' );
  *   useEffect( callback , [ callback, otherDependency ] );
  *
- * @param {Function} getMessage returns confirmation message string if one should appear
- * @return {Function} The callback to execute
+ * @param {string} message the confirmation message string if one should appear
+ * @return {Function} A setter for the navigation message
  */
-const useConfirmNavigation = ( getMessage ) => {
-	const savedCallback = useRef();
+const useConfirmNavigation = () => {
+	const [ message, setMessage ] = useState( '' );
 
```

Comment:
I thought it'd be simpler to manage the state in this component, and just update the handler every time the message is changed. To that end we return the `setMessage` function so components can change the message displayed when navigating away from the current page.

Note the dependencies array that makes it so that the `handler` is updated every time the `message` state is changed.